com.github.camel-tooling.lsp.eclipse.client.tests.ui/tests/com.github.cameltooling.lsp.ui.tests/src/com/github/cameltooling/lsp/ui/tests/CamelLSPCompletionTest.java

```diff
@@ -0,0 +1,254 @@
+		tryEndpointOptionsCompletion();
+
+		cursorPosition = editor.getText().indexOf("<to");
+		editor.setCursorPosition(cursorPosition += 42);
+		tryEndpointOptionsCompletion();
+
+		LogChecker.assertNoLSPError();
+	}
+
+	/**
+	 * Tests code completion is working for additional endpoint options (the part after "&")
+	 */
+	@Test
+	public void testAdditionalEndpointOptions() {
+		editor = new SourceEditor();
+
+		cursorPosition = editor.getText().indexOf("<from");
+		editor.setCursorPosition(cursorPosition += 42);
+		tryAdditionalOptionsCompletion();
+
+		cursorPosition = editor.getText().indexOf("<to");
+		editor.setCursorPosition(cursorPosition += 42);
+		tryAdditionalOptionsCompletion();
+
+		LogChecker.assertNoLSPError();
+	}
+
+	/**
+	 * Tests duplicate endpoint options are filtered out
+	 */
+	@Test
+	public void testDuplicateOptionsFiltering() {
+		editor = new SourceEditor();
+
+		cursorPosition = editor.getText().indexOf("<from");
+		editor.setCursorPosition(cursorPosition += 5);
+		editor.selectText(cursorPosition, cursorPosition + 38);
+		editor.insertText(INSERT_SPACE);
+
+		assertDuplicateOptionsFilltering(Options.ID.option);
+		assertDuplicateOptionsFilltering(Options.URI.option);
+		assertDuplicateOptionsFilltering(Options.REF.option);
+
+		cursorPosition = editor.getText().indexOf("<to");
+		editor.setCursorPosition(cursorPosition += 3);
+		editor.selectText(cursorPosition, cursorPosition + 42);
+		editor.insertText(INSERT_SPACE);
+
+		assertDuplicateOptionsFilltering(Options.ID.option);
+		assertDuplicateOptionsFilltering(Options.URI.option);
+		assertDuplicateOptionsFilltering(Options.REF.option);
+
+		LogChecker.assertNoLSPError();
+	}
+	
+	/**
+	 * Create new empty XML file
+	 */
+	private static void createNewEmptyXMLFile(String fileName) {
+		NewXMLFileWizard newWizXML = new NewXMLFileWizard();
+		newWizXML.open();
+		NewXMLFileFirstPage newWizXMLPage = new NewXMLFileFirstPage(newWizXML);
+		newWizXMLPage.setFileName(fileName);
+		newWizXML.finish();
+	}
+
+	/**
+	 * Prepare testing project
+	 */
+	private static void createNewJavaProject(String name) {
+		JavaProjectWizard javaWiz = new JavaProjectWizard();
+		javaWiz.open();
+		JavaProjectWizardFirstPage javaWizPage = new JavaProjectWizardFirstPage(javaWiz);
+		javaWizPage.setName(name);
+		javaWiz.finish(TimePeriod.DEFAULT);
+	}
+
+	private void assertComponentSchemes(List<String> proposals) {
+		AbstractWait.sleep(TimePeriod.MEDIUM);
+		collector.checkThat("Content assistant is empty", proposals.isEmpty(), equalTo(false));
+		collector.checkThat("Content assistant is not filtered", proposals.get(0).startsWith("f"), equalTo(true));
+	}
+
+	private void assertDuplicateOptionsFilltering(String proposal) {
+		collector.checkThat(proposal, editor.getCompletionProposals().contains(proposal), equalTo(true));
+		selectCompletionProposal(proposal);
+		collector.checkThat(proposal, editor.getCompletionProposals().contains(proposal), equalTo(false));
```

Comment:
hum I think that what is checked is this case:
![image](https://user-images.githubusercontent.com/1105127/47779222-e95d3500-dcf8-11e8-9aed-56d1f3392599.png)

as we are on the option itself, we choose to provide the option too. It might be discussed but it is the expected behavior for now.
What needs to be checked is when teh completion is called on another options, i;e. when it is seprated by a '&' in Java and '&amp;' in xml.
like this case:
![image](https://user-images.githubusercontent.com/1105127/47779365-3a6d2900-dcf9-11e8-9128-7cafedecd59e.png)

Reply:
Thanks for your review @apupier, I did changes inside UI test which will cover use case you described.